### PR TITLE
Simple cart

### PR DESCRIPTION
**Pull request by FrankOelsch**
New page Cart, 
New styled components Item and TextInput, 
fuzzy search, 
routing to cart

### CODE REVIEW

**Review by mpagels — CHANGES_REQUESTED** [review]
Found some semantically issues and left some questions. Pls review.

File: src/components/Item.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import styled from "styled-components";
+
+export default function Item({ id, name, price, inCart, onSelect }) {
+  return (
+    <StyledH3 variante={inCart} id={id} onClick={(e) => onSelect(e)}>
```
[the remark this answers is not in the document]
Comment (FrankOelsch) [reply src/components/Item.js:5]: Right, I do not need, have deleted it.

File: src/components/input/TextInput.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import styled from "styled-components";
+
+export default function TextInput(props) {
+  const onFocus = (e) => {
+    if (e.which === 9) {
+      return false;
```
[the remark this answers is not in the document]
Comment (FrankOelsch) [reply src/components/input/TextInput.js:6]: Yes, event. Saw that 'which' should not be used, used 'key' instead.

File: src/pages/Cart.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  const falseArray = [];
+  const cartArray = [];
+
+  useEffect(() => {
+    data.forEach((item) => {
+      if (item.id === toggleID) {
+        item.inCart = !item.inCart;
+        setToggleID("");
+      }
+    });
```
[the remark this answers is not in the document]
Comment (FrankOelsch) [reply src/pages/Cart.js:57]: It should only work in principle, nothing is stored yet, comes only with the next userstory (localStorage).

File: src/pages/Cart.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+            id={item.id}
+            name={item.name}
+            price={item.price}
+            inCart={item.inCart}
+            onSelect={toggleSelect}
+          />
+        ))}
+      </section>
+
+      <StyledH2>Artikel-Suche:</StyledH2>
```
[the remark this answers is not in the document]
Comment (FrankOelsch) [reply src/pages/Cart.js:97]: Right, changed it to 'article'.

[the remark this answers is not in the document]
Comment (FrankOelsch) [reply src/pages/Cart.js:97]: Thank you for the review!

File: src/pages/Cart.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (FrankOelsch) [reply src/pages/Cart.js:57]: Strange, now the toggle suddenly does not work anymore...

[the remark this answers is not in the document]
Comment (FrankOelsch) [reply src/pages/Cart.js:57]: Found my thinking error. Now it works.

File: src/pages/Cart.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+  const falseArray = [];
+  const cartArray = [];
+
+  useEffect(() => {
+    data.forEach((item) => {
+      if (item.id === toggleID) {
+        item.inCart = !item.inCart;
+        setToggleID("");
+      }
+    });
```
Comment (Kheirah) [inline src/pages/Cart.js:57]:
```suggestion
    if (!toggleID) return;

    setData(
      data.map((item) => {
        if (item.id === toggleID) {
          setToggleID("");
          return { ...item, inCart: !item.inCart };
        } else {
          return item;
        }
      })
    );
    }

```
Martin is right in pointing out that React does not "think" that way. Whenever you want to change state, you need to use the respective setState method.

**Pull request merged**